include/avk/cpp_utils.hpp

```diff
@@ -1161,85 +1161,144 @@ namespace avk
 
 #pragma region swap + dispose and handle help functions
 
-	/**
-	*	This concept captures those objects which have a size() method, like vectors	*	
-	*/
+	/**  This concept captures those objects which have a size() method, like vectors
+	 */
 	template <typename T>
 	concept has_size = requires (T x)
 	{
 		x.size();
 	};
 
-	/**
-	*	This concept captures those objects that are explicitely convertible to boolean
-	*  
-	*	Note: most vulkan resources contain an explicit boolean converter
-	*/
+	/** This concept captures those objects that are explicitely convertible to boolean
+	 *
+	 *	Note: most vulkan resources contain an explicit boolean converter.
+	 */
 	template <class T>
 	concept boolean_convertible = requires(T(&f)()) {
 		static_cast<bool>(f());
 	};
 
-	/**
-	* Swaps aOld and aNew and handles the life time of aOld
-	*
-	* The purpose of the functions in this section swap and handle lifetime
-	* in a more clean structure in the source code.
-	*
-	* Additionally, when possible, lifetime_handler+swap is ignored if the
-	* rhs parameter contains no values.
-	*
-	* Be aware that inital value  of aOld  will be disposed of
-	*
-	* after the call, aOld's memory will contain initial aNew's value
-	*
-	* @param aNew				new Resource which will be moved to old Resource's place after operation
-	* @param aOld				old Resource object will be disposed of
-	* @param aLifeTimeHandler	Life time handler for aOld
-	*
-	*/
+	/** Swaps aOld and aNew and handles the life time of aOld.
+	 *
+	 *  Be aware that inital value of aOld will be disposed of.
+	 *  Ater the execution: 1) The reference aNew will be in an undefined state after the execution.
+	 *                      2) aOld's memory will contain initial aNew's value.
+	 *
+	 *  Be aware that inital value of aOld will be disposed of depending on the functionality of aLifeTimeHandler.
+	 *
+	 *  @param aNew				New Resource which will be moved to old Resource's place after operation
+	 *  @param aOld				Old Resource object to be disposed of
+	 *  @param aLifeTimeHandler	Life time handler for aOld
+	 */
 	template<typename T, typename F>
 	void emplace_and_handle_previous(T& aNew, T&& aOld, F&& aLifeTimeHandler)

```

Comment:
Applied the changes as discussed except two difference:
- I kept the reference to swap instead of assign. The more I thought about it, the more I was convinced that some context will be lost otherwise.
- Went with the reference to `lifetime_handle` as you suggested. I think, again, some context will be gained by doing so.

Otherwise the function is kept as we discussed, with a new name.
Please let me know if you disagree with the above.

Reply:
Additionally I would like to keep using std::swap given the name. I think there is a quick "got it" moment when seeing familiar functions.